ndmtestpackage.py

```diff
@@ -0,0 +1,386 @@
+    while ndmPod == None and i < 5:
+        sleep(1)
+
+        # List pods
+        # Assumption: NDM pod runs under 'dafault' namespace.
+        pods = v1.list_namespaced_pod('default')
+
+        # Find NDM Pod
+        for pod in pods.items:
+            #Assumption: Pod name starts with string 'node-disk-manager'.
+            if pod.metadata.name.startswith('node-disk-manager'):
+                ndmPod = pod
+                break
+        i += 1
+    else:
+        if ndmPod == None:
+            print 'Failed getting NDM-Pod.'
+            exit(1)
+
+    return ndmPod
+
+def get_pod_phase(pod):
+    return pod.status.phase
+
+def get_container_state_in_ndm_pod(container_index=0, wait_for_sec=10):
+    waited = 0
+    pod = get_ndm_pod()
+    while pod.status.container_statuses is None and waited < wait_for_sec:
+        sleep(1)
+        waited += 1
+        pod = get_ndm_pod()
+    if not waited < wait_for_sec:
+        raise NDMTestException('Pod had no container till '\
+                               + str(wait_for_sec) + ' seconds.')
+
+    while len(pod.status.container_statuses) <= container_index\
+            and waited < wait_for_sec:
+        sleep(1)
+        waited += 1
+        pod = get_ndm_pod()
+    if not waited < wait_for_sec:
+        raise NDMTestException('Pod did not had ' + str(container_index+1)\
+                               + ' containers till ' + str(wait_for_sec)\
+                               + ' seconds.')
+
+    return pod.status.container_statuses[container_index].state
+
+def make():
+    # Run make command
+    try:
+        subprocess.check_call("make")
+    except subprocess.CalledProcessError as err:
+        print 'Subprocess error occured while `make`ing:', err.returncode
+        raise err
+    except Exception as err:
+        print 'Unknown error occured while `make`ing.'
+        raise err
+
+def yaml_prepare():
+    # Prepare the yaml
+    with open(NDM_YAML) as f:
+        ndmYaml = yaml.load(f)
+
+    # Assumption: In following two statements it is assumed that
+    # this pod has only one container
+
+    # Set image name
+    ndmYaml['spec']['template']['spec']['containers'][0]['image']\
+        = DOCKER_IMAGE_NAME + ":" + DOCKER_IMAGE_TAG
+
+    # set imagePullPolicy
+    ndmYaml['spec']['template']['spec']['containers'][0]['imagePullPolicy']\
+        = "IfNotPresent"
+
+    with open(NDM_TEST_YAML_PATH+NDM_TEST_YAML_NAME, 'w') as f:
+        yaml.dump(ndmYaml, f)
+
+def yaml_apply():
+    # Apply the yaml
+    try:
+        subprocess.check_call(["kubectl", "apply", "-f",
+                               NDM_TEST_YAML_PATH+NDM_TEST_YAML_NAME])
+    except subprocess.CalledProcessError as err:
+        print 'Subprocess error occured while applying the prepared YAML:',\
+            err.returncode
+        raise err
+    except Exception as err:
+        print 'Unknown error occured while applying the prepared YAML.'
+        raise err
+
+def start_minikube_and_print_output():
```

Comment:
Have a different module for minikube (say minikube-adm.py) for administring minikube functionality like:
- setup
- check status
- clear containers
- teardown